specs/Standards/MultiToken/README.md

```diff
@@ -0,0 +1,539 @@
+    ///   standard. If it does not, MultiToken contract's `mt_resolve_transfer` MUST deal
+    ///   with the resulting failed cross-contract call and roll back the transfer.
+    /// * Contract MUST implement the behavior described in `mt_resolve_transfer`
+    ///
+    /// Arguments:
+    /// * `receiver_id`: the valid NEAR account receiving the token.
+    /// * `token_id`: the token to send.
+    /// * `amount`: amount of tokens to transfer for token_id
+    /// * `memo` (optional): for use cases that may benefit from indexing or
+    ///    providing information for a transfer.
+    /// * `msg`: specifies information needed by the receiving contract in
+    ///    order to properly handle the transfer. Can indicate both a function to
+    ///    call and the parameters to pass to that function.
+    fn mt_transfer_call(
+        &mut self,
+        receiver_id: AccountId,
+        token_id: TokenId,
+        amount: U128,
+        memo: Option<String>,
+        msg: String,
+    ) -> PromiseOrValue<U128>;
+
+    /// Batch token transfer. Transfer a tokens given token_ids and amounts. The token ids can correspond to  
+    /// either Non-Fungible Tokens or Fungible Tokens or some combination of the two. The token ids
+    /// are used to segment the types on a per contract implementation basis.
+    ///
+    /// Requirements
+    /// * Caller of the method must attach a deposit of 1 yoctoⓃ for security purposes
+    /// * Contract MUST panic if called by someone other than token owner or,
+    ///   if using Approval Management, one of the approved accounts
+    /// * `approval_id` is for use with Approval Management,
+    ///   see https://nomicon.io/Standards/NonFungibleToken/ApprovalManagement.html
+    /// * If using Approval Management, contract MUST nullify approved accounts on
+    ///   successful transfer.
+    /// * TODO: needed? Both accounts must be registered with the contract for transfer to
+    ///   succeed. See see https://nomicon.io/Standards/StorageManagement.html
+    /// * The token_ids vec and amounts vec must be of equal length and equate to a 1-1 mapping
+    ///   between amount and id. In the event that they do not line up the call should fail
+    ///
+    /// Arguments:
+    /// * `receiver_id`: the valid NEAR account receiving the token
+    /// * `token_ids`: the tokens to transfer
+    /// * `amounts`: the amount of tokens to transfer for corresponding token_id
+    /// * `approval_ids`: expected approval ID. A number smaller than
+    ///    2^53, and therefore representable as JSON. See Approval Management
+    ///    standard for full explanation. Must have same length as token_ids
+    /// * `memo` (optional): for use cases that may benefit from indexing or
+    ///    providing information for a transfer
+
+    fn mt_batch_transfer(
+        &mut self,
+        receiver_id: AccountId,
+        token_ids: Vec<TokenId>,
+        amounts: Vec<U128>,
+        memo: Option<String>,
+    );
```

Comment:
Is there no desire to be able to transfer to multiple receivers with the same batch transfer call?
If there is a use-case for this, but the initial spec here doesn't incorporate it, should we name these methods differently in some a way so that there is 'room' in the API to add a batch transfer method later that supports multiple recipients?

Reply:
Not sure here. I think alot of the concern with that is gas price. Outside of that I think that a different prefix might be used some thing like mt_multi_batch_transfer. 

Batch transfer is used to try and line up with erc-1155 and aid in developer familiarity.